pkg/build/controller/controller.go

```diff
@@ -302,55 +302,56 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	nextStatus := build.Status.Phase
 	currentReason := build.Status.Reason
 
-	switch pod.Status.Phase {
-	case kapi.PodRunning:
-		// The pod's still running
-		build.Status.Reason = ""
-		build.Status.Message = ""
-		nextStatus = buildapi.BuildPhaseRunning
-
-	case kapi.PodPending:
-		build.Status.Reason = ""
-		build.Status.Message = ""
-		nextStatus = buildapi.BuildPhasePending
-		if secret := build.Spec.Output.PushSecret; secret != nil && currentReason != buildapi.StatusReasonMissingPushSecret {
-			if _, err := bc.SecretClient.Secrets(build.Namespace).Get(secret.Name); err != nil && errors.IsNotFound(err) {
-				build.Status.Reason = buildapi.StatusReasonMissingPushSecret
-				build.Status.Message = buildapi.StatusMessageMissingPushSecret
-				glog.V(4).Infof("Setting reason for pending build to %q due to missing secret %s/%s", build.Status.Reason, build.Namespace, secret.Name)
+	if build.Status.Phase != buildapi.BuildPhaseFailed {
```

Comment:
> Because the retryBuildStatusUpdate (https://github.com/openshift/origin/blob/master/pkg/build/builder/common.go#L183) was not setting the build phase. So when it encountered an issue updating and had to retry, it was not overwriting the build phase with it's failed phase, so it wasn't hitting my new if statement and skipping the switch/case statement.

ah, ok.  so it was just never getting set properly into etcd in the first place.